swagger.yaml

```diff
@@ -0,0 +1,228 @@
+  /hierarchies/{instance_id}/{dimension_name}}/{code_id}:
+    parameters:
+      - $ref: '#/parameters/instance_id'
+      - $ref: '#/parameters/dimension_name'
+      - $ref: '#/parameters/code_id'
+    get:
+      summary: "Get a description of a filter job"
+      description: "Get document describing the filter job"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "The filter job was found and document is returned"
+          schema:
+            $ref: '#/definitions/CodeResponse'
+        400:
+          $ref: '#/responses/InstanceOrDimensionNotFound'
+        404:
+          $ref: '#/responses/CodeNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+responses:
+  InstanceNotFound:
+    description: "Instance not found"
+  DimensionNotFound:
+    description: "Dimension not found"
+  InstanceOrDimensionNotFound:
+    description: "Instance or dimension name not found"
+  CodeNotFound:
+    description: "Code not found"
+  InternalError:
+    description: "Failed to process the request due to an internal error"
+definitions:
+  HierarchyResponse:
+    description: ""
+    type: object
+    properties:
+      links:
+        $ref: '#/definitions/HierarchyLinks'
+      children:
+        type: array
+        description: "The top level nodes of this hierarchy"
+        items:
+          type: object
+          items:
+              $ref: '#/definitions/HierarchyChild'
+  HierarchyLinks:
+    description: "A list of links related to this hierarchy"
+    readOnly: true
+    type: object
+    properties:
+      self:
+        $ref: '#/definitions/SelfLink'
+      dimension:
+        $ref: '#/definitions/DimensionLink'
+  HierarchyChild:
+    type: object
+    readOnly: true
+    properties:
+      links:
+        $ref: '#/definitions/HierarchyChildLinks'
+      label:
+        type: string
+        description: "The label for this child node"
+      code:
+        type: string
+        description: "The code for this child node"
+      no_of_children:
+        type: integer
+        description: "The number of child nodes that this node has"
+  HierarchyChildLinks:
+    description: "A list of links related to this node"
+    readOnly: true
+    type: object
+    properties:
+      self:
+        $ref: '#/definitions/SelfLink'
+  CodeResponse:
+    description: ""
+    type: object
+    properties:
+      links:
+        $ref: '#/definitions/CodeLinks'
+      label:
+        type: string
+        description: "The label for this node"
+      code:
+        type: string
+        description: "The code for this node"
+      has_data:
+        type: boolean
+        description: "True if the instance has an observation for this code"
+      children:
+        type: array
+        description: "The child nodes of this node in the hierarchy"
+        items:
+          type: object
+          items:
+            $ref: '#/definitions/CodeChild'
```

Comment:
This is not coming through on the example response for the endpoint - feels like it should be but not sure why - I might be missing something that explains why it actually shouldnt.

Reply:
I have added some example data for child nodes